src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2757,7 +2757,12 @@ public override InterfaceMapping GetInterfaceMap([DynamicallyAccessedMembers(Dyn
                 MethodBase? rtTypeMethodBase = GetMethodBase(reflectedType, classRtMethodHandle);
                 // a class may not implement all the methods of an interface (abstract class) so null is a valid value
                 Debug.Assert(rtTypeMethodBase is null || rtTypeMethodBase is RuntimeMethodInfo);
-                im.TargetMethods[i] = (MethodInfo)rtTypeMethodBase!;
+                MethodInfo targetMethod = (MethodInfo)rtTypeMethodBase!;
+                // https://github.com/dotnet/runtime/issues/90863
+                if (targetMethod.IsGenericMethod && !targetMethod.IsGenericMethodDefinition)
```

Comment:
A comment a few lines above says that `targetMethod` can be null. We will crash with null reference exception if that happens. Are we missing a test for it?

Reply:
Good question, I'll just make this null safe.